src/ruby/lib/grpc/google_rpc_status_utils.rb

```diff
@@ -28,8 +25,14 @@ class GoogleRpcStatusUtils
     def self.extract_google_rpc_status(status)
       fail ArgumentError, 'bad type' unless status.is_a? Struct::Status
       grpc_status_details_bin_trailer = 'grpc-status-details-bin'
-      return nil if status.metadata[grpc_status_details_bin_trailer].nil?
-      Google::Rpc::Status.decode(status.metadata[grpc_status_details_bin_trailer])
+      binstatus = status.metadata[grpc_status_details_bin_trailer]
+      return nil if binstatus.nil?
+
+      # Lazily load grpc_c and protobuf_c.so for users of this method.
+      require_relative './grpc'
```

Comment:
nit: I think we can remove the corresponding logic from the caller of this in `errors.rb`

Reply:
Agreed. Updated.